### PR TITLE
Make DnsHandle::send &self instead of &mut self

### PR DESCRIPTION
**Pull request by ibigbug**
following #1933 

making the trait non mut so usage of it doesn't need Mutex

### CODE REVIEW

**Comment by ibigbug** [comment]
cc @djc

**Comment by bluejekyll** [comment]
wow, I'm amazed this required so few changes. This has probably been `&mut` since the first non-async implementations. This should definitely make the interface simpler for folks to use. I'll review in more detail once the tests are passing.

**Comment by ibigbug** [comment]
thank you @bluejekyll for the awesome project it's definetely help me a lot in building my project and hope i can contribute some too

**Comment by ibigbug** [comment]
seems coverage is failing for some reason

**Comment by bluejekyll** [comment]
Coverage is for reference and to draw attention to areas where we might not have great coverage so we should take more care on the review. So don't worry about those.

@djc, I'm ok with this change, though it will mean that with out this being mut, if for some reason we need it to be in the future, it will be a harder and bigger breaking change. I don't know whey that would be, but that is the only negative I see to this.

File: crates/proto/src/xfer/dnssec_dns_handle.rs [hunk 1]
```diff
@@ -715,7 +715,7 @@ where
         .filter_map(|rrsig|rrsig.into_data())
         .map(|sig| {
             let rrset = Arc::clone(&rrset);
-            let mut handle = handle.clone_with_context();
+            let  handle = handle.clone_with_context();
```
Comment (djc) [inline crates/proto/src/xfer/dnssec_dns_handle.rs:718]: Nit: there seems to be a double space here? Not sure why rustfmt isn't catching it.

**Review by djc — APPROVED** [review]
LGTM, modulo one tiny nit.

File: crates/proto/src/xfer/dnssec_dns_handle.rs (hunk 1 above)
Comment (ibigbug) [inline crates/proto/src/xfer/dnssec_dns_handle.rs:718]:
```suggestion
            let handle = handle.clone_with_context();
```

**Comment by djc** [comment]
Thanks for fixing that up -- and should have squashed this during merge, sorry for messing that up.

**Pull request merged**